### PR TITLE
Added search and filters for api

### PR DESCRIPTION
**Pull request by sorinburghiu2323**
Check the API doc to find out how to use those spicy filters.

### CODE REVIEW

File: backend/Controllers/community_controller.py [hunk 1]
```diff
@@ -160,6 +161,15 @@ def list_communities(request):
             communitymember__user=user
         ).order_by("name")
 
+    # Check for filter phrase.
+    phrase = request.GET.get("phrase")
+    if phrase is not None:
+        if phrase != "" and not phrase.isspace():
+            for term in phrase.split():
+                communities = communities.filter(Q(name__icontains=term))
```
Comment (milokhiggins) [inline backend/Controllers/community_controller.py:169]: Getting a NameError here, have you imported Q?

**Review by milokhiggins — CHANGES_REQUESTED** [review]
looks good, just missing one import.

File: backend/Controllers/community_controller.py (hunk 1 above)
Comment (sorinburghiu2323) [reply backend/Controllers/community_controller.py:169]: Yes its imported on line 1 right?

Comment (WilliamOrringe) [reply backend/Controllers/community_controller.py:169]: it works

**Review by WilliamOrringe — APPROVED** [review]

**Pull request merged**